packages/fx-core/src/plugins/resource/blazor/plugin.ts

```diff
@@ -0,0 +1,309 @@
+      DependentPluginInfo.botPassword
+    );
+
+    this.pushAppSettings(site, AppSettingsKey.botId, botId);
+    this.pushAppSettings(site, AppSettingsKey.botPassword, botPassword);
+  }
+
+  public collectAppSettingsFromAAD(site: Site, aadConfig: ReadonlyPluginConfig) {
+    const clientId: string = this.checkAndGet(
+      aadConfig.get(DependentPluginInfo.clientID) as string,
+      DependentPluginInfo.clientID
+    );
+    const clientSecret: string = this.checkAndGet(
+      aadConfig.get(DependentPluginInfo.aadClientSecret) as string,
+      DependentPluginInfo.aadClientSecret
+    );
+    const oauthHost: string = this.checkAndGet(
+      aadConfig.get(DependentPluginInfo.oauthHost) as string,
+      DependentPluginInfo.oauthHost
+    );
+    const tenantId: string = this.checkAndGet(
+      aadConfig.get(DependentPluginInfo.tenantId) as string,
+      DependentPluginInfo.tenantId
+    );
+    const applicationIdUris: string = this.checkAndGet(
+      aadConfig.get(DependentPluginInfo.applicationIdUris) as string,
+      DependentPluginInfo.applicationIdUris
+    );
+
+    this.pushAppSettings(site, AppSettingsKey.clientId, clientId);
+    this.pushAppSettings(site, AppSettingsKey.clientSecret, clientSecret);
+    this.pushAppSettings(site, AppSettingsKey.oauthHost, `${oauthHost}/${tenantId}`);
+    this.pushAppSettings(site, AppSettingsKey.identifierUri, applicationIdUris);
+    this.pushAppSettings(
+      site,
+      AppSettingsKey.aadMetadataAddress,
+      AzureInfo.aadMetadataAddress(tenantId)
+    );
+  }
+
+  public pushAppSettings(site: Site, newName: string, newValue: string, replace = true): void {
+    if (!site.siteConfig) {
+      site.siteConfig = {};
+    }
+
+    if (!site.siteConfig.appSettings) {
+      site.siteConfig.appSettings = [];
+    }
+
+    const kv: NameValuePair | undefined = site.siteConfig.appSettings.find(
+      (kv) => kv.name === newName
+    );
+    if (!kv) {
+      site.siteConfig.appSettings.push({
+        name: newName,
+        value: newValue,
+      });
+    } else if (replace) {
+      kv.value = newValue;
+    }
+  }
+
+  public async deploy(ctx: PluginContext): Promise<TeamsFxResult> {
+    Logger.info(Messages.StartDeploy(PluginInfo.displayName));
+
+    this.syncConfigFromContext(ctx);
+
+    const workingPath = ctx.root;
+    const webAppName = this.checkAndGet(this.config.webAppName, ConfigKey.webAppName);
+    const resourceGroupName = this.checkAndGet(
+      this.config.resourceGroupName,
+      ConfigKey.resourceGroupName
+    );
+    const subscriptionId = this.checkAndGet(this.config.subscriptionId, ConfigKey.subscriptionId);
+    const credential = this.checkAndGet(
+      await ctx.azureAccountProvider?.getAccountCredentialAsync(),
+      ConfigKey.credential
+    );
+
+    // ? Do we support user customize build? If yes, how?
+    // * If we support user customize runtime, we need to validate its value because we use it to concat build command.
```

Comment:
I think the runtime is bounded to the azure resource's env. So I think we don't need to support custom now. Or we can retrieve it from the azure resource's config.